src/wallet/wallet2.cpp

```diff
@@ -2197,6 +2334,9 @@ crypto::secret_key wallet2::generate(const std::string& wallet_, const std::stri
   generate_genesis(b);
   m_blockchain.push_back(get_block_hash(b));
 
+  m_subaddress_labels.clear();
+  add_subaddress_account(tr("Primary account"));
+  m_current_subaddress_account = 0;
```

Comment:
Actually no, `add_subaddress_account() ` must be called after the account key is initialized/loaded (otherwise subaddresses cannot be generated). Since `clear()` is called prior to the account key initialization, `add_subaddress_account()` shouldn't be called from `clear()`. Clearing other data (hashtable and labels) should also be done outside `clear()`, because `clear()` is called from `rescan_blockchain()` and we don't want to lose all the subaddress label data every time the user issues the `rescan_bc` command.

I find it quite surprising that `rescan_bc` currently deletes other unrecoverable information (tx secret keys and destination addresses). Is this behavior really intentional?
https://github.com/monero-project/monero/issues/2177